### DISCUSSION TITLE
Variables in Switch-Case

### DISCUSSION THREAD

**Discussion by jnsprnw** [ivanhofer/typesafe-i18n#720] [Q&A] [answered]
Hey!
Thank you for the wonderful library!

Is it possible to use variables in switch-cases? I want to implement something in a similar way like this:
```js
const text = 'I’m {age:number} years old{hasRegion|{true: from {region}, false:}}.';
const str = LLL(test, { age: 10, hasRegion: true, region: 'Asia' });
```
Expected output: `I’m 10 years old from Asia.`

```js
const str = LLL(test, { age: 10, hasRegion: false });
```
Expected output: `I’m 10 years old.`

The output I get at the moment is `I’m 10 years old{hasRegion|}.`.

Is that possible at all?

**Comment by ivanhofer** [comment, marked as answer]
Hi, thanks for your kind words :)

No, this is currently not possible.

It is not the prettiest code, but you could do it like this:

```ts
const text = 'I’m {age:number} years old {appendix:string}'
const textAppendix = 'from {region}'
const region = 'Asia'
const str = LLL(text, { age: 11, appendix: region ? LLL(textAppendix, { region }) : '' }).trim()
```

Do you really use the raw `LLL` function with plain strings instead of the `LL` function with an dictionary object?

**Reply by jnsprnw** [reply]
Ah, I see. I guess, it’s easier to just use two different templates depending on `hasRegion`.

And no, I’m using `LL`. This was just to quickly test multiple templates.
